x/crosschain/keeper/msg_server_vote_outbound.go

```diff
@@ -136,18 +138,105 @@ func (k msgServer) VoteOutbound(
 	return &types.MsgVoteOutboundResponse{}, nil
 }
 
-// FundStabilityPool funds the stability pool with the remaining fees of an outbound tx
-// The funds are sent to the gas stability pool associated with the receiver chain
-// This wraps the FundGasStabilityPoolFromRemainingFees function and logs an error if it fails.We do not return an error here.
+// ManageUnusedGasFee funds the stability pool with the remaining fees of an outbound tx
+// The funds are sent to the gas stability pool associated with the receiver chain and the user is refunded if possible.
+// This wraps the UseRemainingGasFee function and logs an error if it fails.We do not return an error here.
 // Event if the funding fails, the outbound tx is still processed.
-func (k Keeper) FundStabilityPool(ctx sdk.Context, cctx *types.CrossChainTx) {
-	// Fund the gas stability pool with the remaining funds
-	if err := k.FundGasStabilityPoolFromRemainingFees(ctx, *cctx.GetCurrentOutboundParam(), cctx.GetCurrentOutboundParam().ReceiverChainId); err != nil {
-		ctx.Logger().
-			Error("%s: CCTX: %s Can't fund the gas stability pool with remaining fees %s", voteOutboundID, cctx.Index, err.Error())
+func (k Keeper) ManageUnusedGasFee(ctx sdk.Context, cctx *types.CrossChainTx) {
+	outboundParams := cctx.GetCurrentOutboundParam()
+	// If the userGasFeePaid is nil, it means that this cctx was created before the userGasFeePaid was added.
```

Comment:
> this cctx was created before the userGasFeePaid was added
maybe we can add more details in comment about this case

Reply:
could you specify what sort of details would be helpful ?